opm/parser/eclipse/EclipseState/Schedule/MSW/Segment.hpp

```diff
@@ -100,6 +112,13 @@ namespace Opm {
         // indicate if the data related to 'INC' or 'ABS' is ready
         // the volume will be updated at a final step.
         bool m_data_ready;
+        // indicate the type of the segment
+        // regular or spiral ICD
+        SegmentType m_segment_type = SegmentType::REGULAR;
+
+        // information related to SpiralICD. It is nullptr for segments are not
+        // spiral ICD type
+        std::shared_ptr<SpiralICD> m_spiral_icd;
```

Comment:
WSEGSICD will be developed along with WSEGVALV for now .  Due to the time limit, we might not be able to get a base class ICD for them.  When we have a third ICD type join, definitely we should have a ICD base class, then here will be 
```
std::shared_prt<ICD> m_icd;
```

The tricky part will be the interface and what if ICD types changes. 

Everything is still in progress, let us see what will happen.